generic-jms-ra-jar/src/main/java/org/jboss/resource/adapter/jms/inflow/JmsServerSession.java

```diff
@@ -165,7 +165,11 @@ public void onMessage(Message message) {
                 endpoint.afterDelivery();
             }
         } catch (Throwable t) {
-            log.error("Unexpected error delivering message " + message, t);
+            try {
+               log.error("Unexpected error delivering message. " + message != null ? "JMSMessageID " + message.getJMSMessageID() : "Message is null.", t);
```

Comment:
I don't think `message` can ever be `null` in this method (it is only called when there is an actual message to deliver to the consumer

Reply:
I thought the same, but since the spec allows it decided to cover it. It's changed now.